### PR TITLE
Add SupportsSyntaxTree checks to resolve NullReferenceExceptions in Salsa ContainedDocument scenarios

### PR DESCRIPTION
**Pull request by jramsay**
A document can contain multiple ContainedDocuments if both Razor and JS script blocks are present.
When a linked or related document is retrieved a check is required to ensure it supports syntax trees.
Otherwise a NullReferenceException will be hit when there is an attempt to get the root of the syntax tree. @CyrusNajmabadi, @vladima

### CODE REVIEW

File: src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs [hunk 1]
```diff
@@ -52,6 +52,11 @@ public static bool ShouldHideAdvancedMembers(this Document document)
             foreach (var linkedDocumentId in linkedDocumentIds)
             {
                 var linkedDocument = document.Project.Solution.GetDocument(linkedDocumentId);
+                if (!linkedDocument.SupportsSyntaxTree)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs]: I'm not sure about this one.  This seems like a generic item to just process over documents/linked documents.  I don't see anything about it that would be specific to documents with syntax trees.

File: src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs [hunk 2]
```diff
@@ -76,6 +81,11 @@ public static bool ShouldHideAdvancedMembers(this Document document)
             foreach (var linkedDocumentId in document.GetLinkedDocumentIds())
             {
                 var linkedDocument = solution.GetDocument(linkedDocumentId);
+                if (!linkedDocument.SupportsSyntaxTree)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs]: Same with this.  It seems like we'd need to fix callers of this method if we're really having problems.

**Comment by CyrusNajmabadi** [comment]
you'll need to discuss with @davkean  and @Pilchie where this should go depending on what release vehicle you want this to be in.

File: src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs (hunk 1 above)
Comment (jramsay) [inline src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs]: Will send you a screen shot of the exception and call stack to see if there is a more targeted fix.

**Comment by jramsay** [comment]
@davkean and @Pilchie:  I'd like to target this fix for insertion in to ml.  Is this the correct branch or should I be making changes against future?

**Comment by davkean** [comment]
Future is branch you want.

**Comment by jramsay** [comment]
thanks - will make the changes @CyrusNajmabadi noted and send out another PR against Future

File: src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs [hunk 3]
```diff
@@ -91,7 +91,13 @@ public static IEnumerable<Document> GetLinkedDocuments(this Document document)
 
             foreach (var linkedDocumentId in document.GetLinkedDocumentIds())
             {
-                yield return solution.GetDocument(linkedDocumentId);
+                var linkedDocument = solution.GetDocument(linkedDocumentId);
```
Comment (jramsay) [inline src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs]: will revert this one as well

**Comment by jramsay** [comment]
Closing this request and creating a new PR against Future

**Pull request closed without merging**